### PR TITLE
Setup/fix doctests

### PR DESCRIPTION
**Pull request by tkf**

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ makedocs(;
     repo="https://github.com/JuliaObjects/ConstructionBase.jl/blob/{commit}{path}#L{line}",
     sitename="ConstructionBase.jl",
     authors="Takafumi Arakaki, Rafael Schouten, Jan Weidner",
-    assets=String[],
+    strict=true,
```
Comment (jw3126) [inline docs/make.jl]: So this means only build the docs if doctests pass?

Comment (tkf) [reply docs/make.jl]: Yes (or I think more precisely deploy only if doctests pass)

Comment (jw3126) [reply docs/make.jl]: Ok thanks I was not aware that this was off by default.

**Comment by jw3126** [comment]
LGTM

File: docs/make.jl (hunk 1 above)
Comment (tkf) [reply docs/make.jl]: In Documenter 0.24 it looks like we'll have `Documenter.doctest` so we can run doctests more easily via runtests.jl (and stop worrying about a setup like this).

**Pull request merged**